opendatahub-io/data-science-pipelines-operator
# Update go version to 1.20 and resolve CVEs

## The issue resolved by this Pull Request:
Resolves https://issues.redhat.com/browse/RHOAIENG-2260

## Description of your changes:


Bumped go version to 1.20.
Updated K8s package versions to eliminate high level snyk CVE. Verified that the package versions are compatible with each other: https://github.com/kubernetes-sigs/controller-runtime/blob/v0.15.0/go.mod#L20-L27
Also updated the DSPO controller go code to fix all breaking changes introduced by upgrading the `sigs.k8s.io/controller-runtime` package to 0.15.0.

## Testing instructions

Imported my fork in synk and made sure the relevant CVEs were fixed with this update.

Updated package versions locally based on what versions are compatible with each other and ran `go mod tidy`. 
Ran `go build` locally to make sure no errors were introduced due to any breaking changes arising from package version upgrades.
Pushed changes to my fork and ensured the CVEs were resolved.

## Checklist
